libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/WayNameView.java

```diff
@@ -10,12 +10,14 @@
 import androidx.annotation.Nullable;
 import androidx.core.graphics.drawable.DrawableCompat;
 
+import com.mapbox.core.utils.TextUtils;
 import com.mapbox.services.android.navigation.ui.v5.R;
 import com.mapbox.services.android.navigation.ui.v5.ThemeSwitcher;
 
 public class WayNameView extends FrameLayout {
 
   private TextView wayNameText;
+  private boolean visible = false;
```

Comment:
maintain an internal variable which remembers what's the caller's choice: show this view or not.